tests/tests_pytorch/models/test_hooks.py

```diff
@@ -470,18 +470,14 @@ def training_step(self, batch, batch_idx):
         {"name": "Callback.on_sanity_check_start", "args": (trainer, model)},
         {"name": "val_dataloader"},
         {"name": "train", "args": (False,)},
-        {"name": "on_validation_model_eval"},
```

Comment:
Now that these hooks are only called when explicitly overridden by the user, I should extend/parameterize the test to make this conditional. @carmocca Do you have some idea how I could conditionally define an overridden hook on the HookedModel defined here?

Reply:
You can simply override it on the class ([example](https://github.com/Lightning-AI/lightning/blob/feature/eval-mode/tests/tests_pytorch/models/test_hooks.py#L393-L395)) or monkeypatch it on the class if you want to do it based on a pytest parametrization value.

Reply:
I decided to override it by default, and then conditionally set the `model.hook = None` for the exclusion.